### ISSUE TITLE
Windows 7 / 10 - X-Box 360 wireless controller - D-PAD not working

### ISSUE THREAD

**Issue by chrizzlibaer** [supertuxkart/stk-code#2850]
Hi,

i got the issue with 0.9.2 under Windows 7 (i actually set up new Windows 7 to validate) and Windows 10, different PCs.
I got the XBOX 360 Wirless Controller with the USB-Receiver, recognized correctly, working completely in other games like for instance need for speed or broforce.

I can bind the DPAD to (for instance) Steering and navigating through the menu
![binding](https://cloud.githubusercontent.com/assets/28846907/26287244/db7422b2-3e77-11e7-88ef-3f8ab167aa18.PNG)

But I cannot use the binded buttons afterwards, just nothing happens.
I can use any other key, even shoulder buttons and triggers, everything, except the D-PAD.
Even working with multiple controllers - no problem

input Configuration (%appdata%\Supertuxkart) lookls like this:

```
<gamepad name ="Controller (Xbox 360 Wireless Receiver for Windows)" deadzone="4096" desensitize="0" analog="1"
         enabled="true">
    <action name="steerLeft" id="101" event="2" character="0" direction="0" range="0" />
    <action name="steerRight" id="101" event="2" character="0" direction="1" range="0" />
    <action name="accel" id="0" event="3" character="0" />
    <action name="brake" id="3" event="3" character="0" />
    <action name="nitro" id="4" event="3" character="0" />
    <action name="drift" id="5" event="3" character="0" />
    <action name="rescue" id="8" event="3" character="0" />
    <action name="fire" id="1" event="3" character="0" />
    <action name="lookBack" id="6" event="3" character="0" />
    <action name="pauserace" id="9" event="3" character="0" />
    <action name="menuUp" id="100" event="2" character="0" direction="1" range="0" />
    <action name="menuDown" id="100" event="2" character="0" direction="0" range="0" />
    <action name="menuLeft" id="101" event="2" character="0" direction="0" range="0" />
    <action name="menuRight" id="101" event="2" character="0" direction="1" range="0" />
    <action name="menuSelect" id="0" event="3" character="0" />
    <action name="menuCancel" id="3" event="3" character="0" />
</gamepad>
```

It seems like the game is interpreting the DPAD as an analogue stick?
Other Software / Games seem to interprete the DPAD as "POV"?

I even tried to remap the buttons of the controller(s) to keyboard buttons using xmapper (xmapper.com) and the buttons are even recognized in the binding menu.
But then, when I try to start any game, SuperTuxKart crashes with error like "no image", but this shouldnt be the solution anyways I guess...

Any ideas on that?
Thanks in Advance!

**Comment by auriamg** [comment]
has anyone got an Xbox gamepad to test this? I do not own one myself so it's a little hard to investigate

**Comment by hiker** [comment]
Hi,

I can test that. Chrizzlibaer, can you open a ticket 
(https://github.com/supertuxkart/stk-code/issues)? This way you would be 
kept updated.

Cheers,
    Joerg

**Comment by auriamg** [comment]
Never mind the part about creating a ticket, hiker probably replied to a mail without realizing it was already from a github ticket

**Comment by hiker** [comment]
Just realised there already is a ticket, sorry :)

Cheers,
    Joerg

**Comment by hiker** [comment]
Reason is that the HAT is reported as axis 100 and 101 - which is larger than the number of axis on the device. I'm working on a patch.

**Comment by hiker** [comment]
@auriamg Fixed in e6a134c3479ccdbaf91bb36d46ffb64b6a3b52b9. The fix might not be ideal (axis count is increased from say 5 on xbox to 101, though the memory increase is very small. If you have a better idea, let me know, else merge with master for the next release.

**Comment by hiker** [comment]
Accepting this fix as good enough :)
